pkg/opensearch/snapshot_tests/helpers_test.go

```diff
@@ -0,0 +1,80 @@
+package snapshot_tests
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/stretchr/testify/require"
+)
+
+type queryDataTestRoundTripper struct {
+	requestCallback func(req *http.Request) error
+	body            []byte
+	statusCode      int
+}
+
+// we fake the http-request-call. we return a fixed byte-array (defined by the test snapshot),
+// and we also check if the http-request-object has the correct data
+func (rt *queryDataTestRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	err := rt.requestCallback(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return &http.Response{
+		StatusCode: rt.statusCode,
+		Header:     http.Header{},
+		Body:       io.NopCloser(bytes.NewReader(rt.body)),
+	}, nil
+}
+
+func newTestDsSettings() *backend.DataSourceInstanceSettings {
+	return &backend.DataSourceInstanceSettings{
+		JSONData: json.RawMessage(`{
+			"database":"opensearch_dashboards_sample_data_flights",
+			"flavor":"opensearch",
+			"pplEnabled":true,
+			"version":"2.3.0",
+			"timeField":"timestamp",
+			"interval":"Daily",
+			"timeInterval":"1s",
+			"maxConcurrentShardRequests":42
+		}`),
+		Database: "[testdb-]YYYY.MM.DD",
+		URL:      "http://localhost:9200",
+	}
```

Comment:
should we include things like the log level field and log message field in the settings?

Reply:
Good point, looks like I added those fields with the Lucene logs migration. I will add it in the upcoming snapshot test PR for Lucene logs